libical: 3.0.8 -> 3.0.9 (fix build on staging)

Mic92 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Mic92 committed:
libical: 3.0.8 -> 3.0.9

SuperSandro2000 commented on pkgs/development/libraries/libical/default.nix:
```diff
@@ -75,12 +75,6 @@ stdenv.mkDerivation rec {
     ./respect-env-tzdir.patch
-    # Export src-generator binary for use while cross-compiling
-    # https://github.com/libical/libical/pull/439
-    (fetchpatch {
```
Please remove fetchpatch from inputs.

Mic92 replied:
done.

Mic92 committed:
linux: fix DRM_AMD_DC_DCN3_0 beeing defined twice

both https://github.com/NixOS/nixpkgs/pull/111177 and
https://github.com/NixOS/nixpkgs/pull/109105

introduced the same option.

Mic92 commented:
@GrahamcOfBorg eval

Mic92 commented:
ofborg you are drunk! If I compare the lines in the stacktrace they don't line up with the code from this repo!
